Google.Api.Gax.Grpc/GrpcAdapter.cs

```diff
@@ -84,5 +104,29 @@ private static GrpcAdapter DetectDefaultPreferringGrpcNetClient()
                 return GrpcCoreAdapter.Instance;
             }
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="grpcApiDescriptor"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public static GrpcAdapter GetFallbackAdapter(GrpcApiDescriptor grpcApiDescriptor)
+        {
+            // TODO: Some way of indicating a preference? Or just set the adapter in the client builder...?
+            if (grpcApiDescriptor.Transports.HasFlag(GrpcTransports.Grpc))
+            {
+                // TODO: This is all a bit of a mess.
+                return CreateDefaultAdapter();
```

Comment:
Why not?
```suggestion
                return s_defaultFactory.Value;
```

Reply:
Or even 
```suggestion
                return DefaultAdapter;
```

Reply:
Changed to `DefaultAdapter`, but I'm expecting all of this will change anyway, and we won't expose a single default adapter - this method will effectively replace it. But I think we can get to the details of that slightly later in the evolution of this PR :)